application/view/omeka/site/index/search.phtml

```diff
@@ -1,43 +1,51 @@
-<?php echo $this->pageTitle(sprintf($this->translate('Search results for “%s”'), $query)); ?>
+<?php
+$translate = $this->plugin('translate');
+$hyperlink = $this->plugin('hyperlink');
+$url = $this->plugin('url');
+?>
+
+<?php echo $this->pageTitle(sprintf($translate('Search results for “%s”'), $query)); ?>
+
+<?php if (empty($results)): ?>
+    <p><?php echo $translate('No results.'); ?></p>
+<?php else: ?>
+<?php
+$resourceLabels = [
+    'site_pages' => 'Site pages', // @translate
+    'items' => 'Items', // @translate
+    'item_sets' => 'Item sets', // @translate
+    'media' => 'Media', // @translate
+];
+$resourceControllers = [
+    'site_pages' => 'page',
+    'items' => 'item',
+    'item_sets' => 'item-set',
+    'media' => 'media',
+];
+?>
+    <?php foreach ($results as $resourceName => $result): ?>
 <div>
-    <h2><?php echo $this->translate('Pages'); ?></h2>
-    <?php if ($pages): ?>
+    <h2><?php echo $translate($resourceLabels[$resourceName]); ?></h2>
+
+    <?php if ($result['total']): ?>
+        <?php $titleMethod = $resourceName === 'site_pages' ? 'title' : 'displayTitle'; ?>
     <ul>
-        <?php foreach ($pages as $page): ?>
-        <li><?php echo $page->link($page->title()); ?></li>
+        <?php foreach ($result['resources'] as $resource): ?>
+        <li><?php echo $resource->link($resource->$titleMethod()); ?></li>
         <?php endforeach; ?>
     </ul>
-    <?php echo $this->hyperlink(
-        sprintf($this->translate('View all results (%s total)'), $pagesTotal),
-        $this->url(
-            'site/page-browse',
-            ['controller' => 'page', 'action' => 'browse'],
-            ['query' => ['fulltext_search' => $query]],
-            true
-        )
-    ); ?>
-    <?php else: ?>
-    <p><?php echo $this->translate('No pages found'); ?></p>
-    <?php endif; ?>
-</div>
-<div>
-    <h2><?php echo $this->translate('Items'); ?></h2>
-    <?php if ($items): ?>
-    <ul>
-        <?php foreach ($items as $item): ?>
-        <li><?php echo $item->link($item->displayTitle()); ?></li>
-        <?php endforeach; ?>
-    </ul>
-    <?php echo $this->hyperlink(
-        sprintf($this->translate('View all results (%s total)'), $itemsTotal),
-        $this->url(
-            'site/resource',
-            ['controller' => 'item', 'action' => 'browse'],
-            ['query' => ['fulltext_search' => $query]],
-            true
-        )
-    ); ?>
-    <?php else: ?>
-    <p><?php echo $this->translate('No items found'); ?></p>
-    <?php endif; ?>
+        <?php echo $hyperlink(
+            sprintf($translate('View all results (%s total)'), $result['total']),
+            $url(
+                'site/resource',
```

Comment:
So basically, this just happens to work because, even though we have a different route for page browse, the URL it resolves to is the same as the one you get if you use the site/resource route and 'page' as the controller? I believe I'm describing the situation correctly.